invenio_checks/components.py

```diff
@@ -129,5 +142,12 @@ def _run_checks(self, identity, data=None, record=None, errors=None, **kwargs):
                     },
                 )
 
+    def publish(self, identity, data=None, record=None, errors=None, **kwargs):
+        """Run checks on publish."""
+        if errors is None:
+            errors = []
+
+        self._run_checks(identity, data, record, errors)
```

Comment:
I'm not a big fan of the fact that we have one function that "does it all". Can you have a go at splitting `_run_checks` into something we can also start reusing e.g. in the requests "Checks" tab and/or the Request overrides we have in Zenodo for running checks on inclusion requests?

It might be that we actually don't want these to be in this component, but on an API or Service layer of checks.